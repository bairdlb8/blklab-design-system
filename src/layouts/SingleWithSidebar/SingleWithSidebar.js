import React from 'react'
import PropTypes from 'prop-types'

import '../../containers/GridContainer'

import './SingleWithSidebar.scss'

const SingleWithSidebar = ({ className = '', children, sidebarContent }) => {
  return (
    <section className="single-with-sidebar display-grid">
      <article className="pt-lg">{children}</article>
      <aside className="p-lg">{sidebarContent}</aside>
    </section>
  )
}

SingleWithSidebar.propTypes = {
  sidebarContent: PropTypes.any
}

export default SingleWithSidebar
