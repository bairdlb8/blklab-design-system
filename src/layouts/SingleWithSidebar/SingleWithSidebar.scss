@import '../../scss/_mixins';

.single-with-sidebar {
  grid-template-columns: auto 350px;
  gap: 50px;

  article {
  }

  aside {
    background-color: var(--primary-color);
    color: var(--text-color-default);
    box-sizing: border-box;

    @include selectFontColor(var(--text-color-default));
  }
}
