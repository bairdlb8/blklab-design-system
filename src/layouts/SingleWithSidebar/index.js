import SingleWithSidebar from './SingleWithSidebar'
export default SingleWithSidebar
