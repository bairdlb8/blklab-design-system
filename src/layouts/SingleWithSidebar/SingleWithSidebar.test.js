import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { toHaveNoViolations } from 'jest-axe'
import SingleWithSidebar from './SingleWithSidebar'

expect.extend(toHaveNoViolations)

describe('<SingleWithSidebar />', () => {

  describe('accessibility', () => {
    it('should not fail any accessibility tests', async () => {
      const { container } = render(<SingleWithSidebar />)
      expect(await axe(container)).toHaveNoViolations()
    })
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<SingleWithSidebar />)

    expect(asFragment(<SingleWithSidebar />)).toMatchSnapshot()
  })

})
