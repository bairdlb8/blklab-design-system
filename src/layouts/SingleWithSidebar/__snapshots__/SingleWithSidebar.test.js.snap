// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleWithSidebar /> should take a snapshot 1`] = `
<DocumentFragment>
  <section
    class="single-with-sidebar display-grid"
  >
    <article
      class="pt-lg"
    />
    <aside
      class="p-lg"
    />
  </section>
</DocumentFragment>
`;
